### ISSUE TITLE
functorch {vjp, grad, jacrev} parity with autograd.functional.{vjp, jacobian}

### ISSUE THREAD

**Issue by zou3519** [pytorch/functorch#253]
Eventually we'll want to deprecate torch.autograd.functional in favor of the functional functorch APIs. In order to do so, we'll need to add some supported options that autograd.functional.* operations have to functorch.{vjp, grad, jacrev, jvp, jacfwd} transforms.

For example, autograd.functional.jacobian supports the following flags:
- strict=False
- vectorize=False
- create_graph=False

Functorch's jacrev should support:
- strict=False
- vectorize=True (as the default, since it is generally faster)
- No create_graph. Instead, create_graph should be based on whether or not the inputs require gradient and to turn off create_graph when the inputs do require gradient a user should use torch.no_grad like in regular PyTorch.
